gitclone/checkout_method_pr_manual.go

```diff
@@ -8,100 +8,72 @@ import (
 	"github.com/bitrise-io/go-utils/log"
 )
 
-//
+const forkRemoteName = "fork"
+
 // PRManualMergeParams are parameters to check out a Merge Request using manual merge
 type PRManualMergeParams struct {
+	IsFork bool
 	// Source
-	HeadBranch, Commit string
+	HeadBranch  string
+	MergeArg    string
+	HeadRepoURL string // Optional
 	// Target
 	BaseBranch string
 }
 
 //NewPRManualMergeParams validates and returns a new PRManualMergeParams
-func NewPRManualMergeParams(headBranch, commit, baseBranch string) (*PRManualMergeParams, error) {
-	if strings.TrimSpace(headBranch) == "" {
-		return nil, NewParameterValidationError("manual PR merge checkout strategy can not be used: no head branch specified")
+func NewPRManualMergeParams(isFork bool, headBranch, commit, forkRepoURL, baseBranch string) (*PRManualMergeParams, error) {
+	if err := validatePRManualMergeParams(isFork, headBranch, commit, forkRepoURL, baseBranch); err != nil {
+		return nil, err
 	}
-	if strings.TrimSpace(commit) == "" {
-		return nil, NewParameterValidationError("manual PR merge checkout strategy can not be used: no head branch commit hash specified")
-	}
-	if strings.TrimSpace(baseBranch) == "" {
-		return nil, NewParameterValidationError("manual PR merge checkout strategy can not be used: no base branch specified")
+
+	if isFork {
+		remoteForkBranch := fmt.Sprintf("%s/%s", forkRemoteName, headBranch)
+		return &PRManualMergeParams{
+			IsFork:      isFork,
+			HeadBranch:  headBranch,
+			MergeArg:    remoteForkBranch,
+			HeadRepoURL: forkRepoURL,
+			BaseBranch:  baseBranch,
```

Comment:
might the shared properties could be set before this if, inside the if we could set the case-specific parameters, this way we could better highlight the difference

Reply:
Sounds reasonable!